From c0ab0639adc3af9221edc8485e87e29c810f87ad Mon Sep 17 00:00:00 2001
From: Steve Anton <anxuiz.nx@gmail.com>
Date: Fri, 3 Aug 2012 10:07:04 -0700
Subject: [PATCH] Fix PlayerBucketEmptyEvent - canceling will prevent the
 block from forming and the event will also be called for
 creative players.  Fixes BUKKIT-2002 and BUKKIT-1997.

---
 src/main/java/net/minecraft/server/ItemBucket.java |   20 +++++++++-----------
 1 files changed, 9 insertions(+), 11 deletions(-)

diff --git a/src/main/java/net/minecraft/server/ItemBucket.java b/src/main/java/net/minecraft/server/ItemBucket.java
index 4d1e2fc..b1feec1 100644
--- a/src/main/java/net/minecraft/server/ItemBucket.java
+++ b/src/main/java/net/minecraft/server/ItemBucket.java
@@ -134,19 +134,17 @@ public class ItemBucket extends Item {
                     if (!entityhuman.e(i, j, k)) {
                         return itemstack;
                     }
+                    
+                    // CraftBukkit start
+                    PlayerBucketEmptyEvent event = CraftEventFactory.callPlayerBucketEmptyEvent(entityhuman, clickedX, clickedY, clickedZ, movingobjectposition.face, itemstack);
 
-                    if (this.a(world, d0, d1, d2, i, j, k) && !entityhuman.abilities.canInstantlyBuild) {
-                        // CraftBukkit start
-                        PlayerBucketEmptyEvent event = CraftEventFactory.callPlayerBucketEmptyEvent(entityhuman, clickedX, clickedY, clickedZ, movingobjectposition.face, itemstack);
+                    if (event.isCancelled()) {
+                        return itemstack;
+                    }
+                    // CraftBukkit end
 
-                        if (event.isCancelled()) {
-                            return itemstack;
-                        }
-                        // CraftBukkit end
-                        // CraftBukkit TODO: look for all the stuff that disappeared here, and make sure this is still where it should be
-                        // CraftBukkit start
-                        return CraftItemStack.createNMSItemStack(event.getItemStack());
-                        // CraftBukkit end
+                    if (this.a(world, d0, d1, d2, i, j, k) && !entityhuman.abilities.canInstantlyBuild) {
+                        return CraftItemStack.createNMSItemStack(event.getItemStack()); // CraftBukkit
                     }
                 }
             } else if (this.a == 0 && movingobjectposition.entity instanceof EntityCow) {
-- 
1.7.8.msysgit.0

